ComAmazonawsKms/runtimes/python/src/com_amazonaws_kms/smithygenerated/com_amazonaws_kms/aws_sdk_to_dafny.py

```diff
@@ -0,0 +1,2562 @@
+        ),
+        BypassPolicyLockoutSafetyCheck=(
+            Option_Some(input["BypassPolicyLockoutSafetyCheck"])
+            if "BypassPolicyLockoutSafetyCheck" in input.keys()
+            else Option_None()
+        ),
+        Tags=(
+            Option_Some(
+                Seq(
+                    [
+                        com_amazonaws_kms.smithygenerated.com_amazonaws_kms.aws_sdk_to_dafny.com_amazonaws_kms_Tag(
+                            list_element
+                        )
+                        for list_element in input["Tags"]
+                    ]
+                )
+            )
+            if "Tags" in input.keys()
+            else Option_None()
+        ),
+        MultiRegion=(
+            Option_Some(input["MultiRegion"])
+            if "MultiRegion" in input.keys()
+            else Option_None()
+        ),
+    )
+
+
+def com_amazonaws_kms_KeyUsageType(input):
+    # Convert KeyUsageType
+    if input == "SIGN_VERIFY":
+        return KeyUsageType_SIGN__VERIFY()
+    elif input == "ENCRYPT_DECRYPT":
+        return KeyUsageType_ENCRYPT__DECRYPT()
+    else:
+        raise ValueError("No recognized enum value in enum type: " + input)
+
+
+def com_amazonaws_kms_CustomerMasterKeySpec(input):
+    # Convert CustomerMasterKeySpec
+    if input == "RSA_2048":
+        return CustomerMasterKeySpec_RSA__2048()
+    elif input == "RSA_3072":
+        return CustomerMasterKeySpec_RSA__3072()
+    elif input == "RSA_4096":
+        return CustomerMasterKeySpec_RSA__4096()
+    elif input == "ECC_NIST_P256":
+        return CustomerMasterKeySpec_ECC__NIST__P256()
+    elif input == "ECC_NIST_P384":
+        return CustomerMasterKeySpec_ECC__NIST__P384()
+    elif input == "ECC_NIST_P521":
+        return CustomerMasterKeySpec_ECC__NIST__P521()
+    elif input == "ECC_SECG_P256K1":
+        return CustomerMasterKeySpec_ECC__SECG__P256K1()
+    elif input == "SYMMETRIC_DEFAULT":
+        return CustomerMasterKeySpec_SYMMETRIC__DEFAULT()
+    else:
+        raise ValueError("No recognized enum value in enum type: " + input)
+
+
+def com_amazonaws_kms_KeySpec(input):
+    # Convert KeySpec
+    if input == "RSA_2048":
+        return KeySpec_RSA__2048()
+    elif input == "RSA_3072":
+        return KeySpec_RSA__3072()
+    elif input == "RSA_4096":
+        return KeySpec_RSA__4096()
+    elif input == "ECC_NIST_P256":
+        return KeySpec_ECC__NIST__P256()
+    elif input == "ECC_NIST_P384":
+        return KeySpec_ECC__NIST__P384()
+    elif input == "ECC_NIST_P521":
+        return KeySpec_ECC__NIST__P521()
+    elif input == "ECC_SECG_P256K1":
+        return KeySpec_ECC__SECG__P256K1()
+    elif input == "SYMMETRIC_DEFAULT":
+        return KeySpec_SYMMETRIC__DEFAULT()
+    else:
+        raise ValueError("No recognized enum value in enum type: " + input)
+
+
+def com_amazonaws_kms_OriginType(input):
+    # Convert OriginType
+    if input == "AWS_KMS":
+        return OriginType_AWS__KMS()
```

Comment:
What's the reason of doing this conversion?

Reply:
This gets you from the native enum `OriginType` to its Dafny representation
I found this API page which seems to line up with this API: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kms/client/describe_key.html
Interestingly, we appear to be missing `EXTERNAL_KEY_STORE` from our model... https://github.com/aws/aws-cryptographic-material-providers-library/blob/main/ComAmazonawsKms/Model/kms/model.json#L3919-L3937

Reply:
https://sim.amazon.com/issues/CrypTool-5286 for bookkeeping on this